src/chainstate/stacks/db/blocks.rs

```diff
@@ -0,0 +1,3495 @@
+ the Free Software Foundation, either version 3 of the License or
+ (at your option) any later version.
+
+ Blockstack is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY, including without the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with Blockstack. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+use std::io;
+use std::io::prelude::*;
+use std::fmt;
+use std::fs;
+use std::collections::{HashMap, HashSet};
+
+use core::*;
+
+use chainstate::burn::operations::*;
+
+use chainstate::stacks::Error;
+use chainstate::stacks::db::StacksChainState;
+use chainstate::stacks::db::accounts::MinerReward;
+use chainstate::stacks::*;
+use chainstate::stacks::db::*;
+
+use std::path::{Path, PathBuf};
+
+use util::db::Error as db_error;
+use util::db::{
+    DBConn,
+    RowOrder,
+    FromRow,
+    query_rows,
+    query_count,
+    query_int
+};
+
+use util::strings::StacksString;
+
+use util::hash::to_hex;
+
+use chainstate::burn::db::burndb::*;
+
+use net::Error as net_error;
+
+use vm::types::{
+    Value,
+    AssetIdentifier,
+    TupleData,
+    PrincipalData,
+    StandardPrincipalData,
+    QualifiedContractIdentifier
+};
+
+use vm::contexts::{
+    AssetMap
+};
+
+use vm::ast::build_ast;
+use vm::analysis::run_analysis;
+
+use vm::clarity::{
+    ClarityBlockConnection,
+    ClarityInstance
+};
+
+pub use vm::analysis::errors::CheckErrors;
+
+use vm::database::ClarityDatabase;
+
+use vm::contracts::Contract;
+
+use rand::RngCore;
+use rand::thread_rng;
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct StagingMicroblock {
+    pub burn_header_hash: BurnchainHeaderHash,
+    pub anchored_block_hash: BlockHeaderHash,
+    pub microblock_hash: BlockHeaderHash,
+    pub sequence: u16,
+    pub processed: bool,
+    pub orphaned: bool,
+    pub block_data: Vec<u8>
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct StagingBlock {
+    pub burn_header_hash: BurnchainHeaderHash,
+    pub anchored_block_hash: BlockHeaderHash,
+    pub parent_burn_header_hash: BurnchainHeaderHash,
+    pub parent_anchored_block_hash: BlockHeaderHash,
+    pub parent_microblock_hash: BlockHeaderHash,
+    pub parent_microblock_seq: u16,
+    pub microblock_pubkey_hash: Hash160,
+    pub processed: bool,
+    pub attacheable: bool,
+    pub orphaned: bool,
+    pub commit_burn: u64,
+    pub sortition_burn: u64,
+    pub block_data: Vec<u8>
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct StagingUserBurnSupport {
+    pub burn_header_hash: BurnchainHeaderHash,
+    pub anchored_block_hash: BlockHeaderHash,
+    pub address: StacksAddress,
+    pub burn_amount: u64,
+    pub vtxindex: u32,
+}
+
+impl StagingBlock {
+    pub fn is_genesis(&self) -> bool {
+        self.parent_burn_header_hash == FIRST_BURNCHAIN_BLOCK_HASH && self.parent_anchored_block_hash == FIRST_STACKS_BLOCK_HASH
```

Comment:
While working on #1202, I noticed this line. I have a doubt concerning the 
`self.parent_burn_header_hash == FIRST_BURNCHAIN_BLOCK_HASH` part of the test. 
If my understanding is correct, this might be an issue in a real world situation, since *parent_burn_header_hash* of the genesis block will not be the genesis block of bitcoin.. correct?

Reply:
Thanks!  Removed that check; will see if the tests continue to pass.

Reply:
Thanks! Looking at the latest commits, I think there's a similar issue here:
https://github.com/blockstack/blockstack-core/blob/b0b0fb7199929f957f09ef202b74822881d1f0c2/src/chainstate/stacks/db/blocks.rs#L1780

Reply:
Yup, will remove that too (and other instances in the codebase).